server/app.js

```diff
@@ -51,24 +51,25 @@ app.get("/courses/:courseId", (req, res) => {
 // Get All Students In A Course
 app.get("/courses/:courseId/students", (req, res) => {
   const { courseId } = req.params;
-  Student.find({ course: courseId }).then((students) => {
-    if (students.course !== courseId) {
+  Course.findById(courseId).then((course) => {
+    if (course) {
+      Student.find({ course: courseId }).then((students) => {
+        console.log(students);
+        res.json(students);
+        res.status(200);
+      });
+    } else {
       console.log("Not Found!");
       res.json({ error: "Not Found! Please enter a correct courseId" });
-    } else {
-      console.log(students);
-      res.json(students);
-      res.status(200);
     }
   });
 });
-
 // Get Single Student From Specific Course
 app.get("/courses/:courseId/students/:studentId", (req, res) => {
   const { studentId, courseId } = req.params;
-  Student.findById(courseId && studentId)
+  Student.find({ course: courseId, _id: studentId })
```

Comment:
Ha ha, nice solution to that weird task! Usability is questionable, but yeah that was the task :)

Reply:
this was what Namir covered today and I implemented, thanks!